code/game/machinery/newscaster.dm

```diff
@@ -1044,3 +1056,75 @@ GLOBAL_LIST_EMPTY(allCasters)
 			add_fingerprint(user)
 	else
 		return ..()
+
+
+/*
+
+	A small datum for horoscopes (possibly implimented into the newscaster or a pda app?)
+	-candycane/etherware
+
+*/
+
+/datum/horoscope
+
+	var/has_read  // lazylist
+
+	var/static/list/positive = list(
+		"You will find something of great value in maints.",
+		"You will affect the station's balance.",
+		"You will support the station greatly.",
+		"You will come across unexpected glory.",
+		"You will meet a pleasent companion.",
+		"You're about to hit it big."
+	)
+
+	var/static/list/negative = list(
+		"A superior will decieve you,",
+		"You will find great sorrow in maintenance.",
+		"You will be slipped.",
+		"The station will be irriplacably damaged.",
+		"Your life hangs in the balance.",
+		"You will be betrayed by someone close to you."
+	)
+
+	/// if set, will always be returned by get_horoscope
+	var/admin_msg
+	/// the amount your admin mood will be worth
+	var/admin_mood_amount = 1
+
+/datum/horoscope/New()
+	RegisterSignal(SSdcs, COMSIG_NEW_DAY, .proc/reset_readers)
+	. = ..()
+
+/datum/horoscope/proc/check_horoscope(mob/living/carbon/human/reader)
+	if(!reader || (reader?.ckey in has_read))
+		return
+	if(!reader.ckey)
+		to_chat(reader, "<span class='warning'>Alright, so I have no idea how you're seeing this. You somehow have no ckey? Please a-help so we can work this out.</span>")
+		return
+	return TRUE
+
+/datum/horoscope/proc/get_horoscope(mob/living/carbon/human/reader)
+	if(!reader)
+		return
+	if(reader.ckey in has_read)
+		return "Check back tomorrow."
+	LAZYADD(has_read, reader?.ckey)
+
+	if(admin_msg)
+		var/positive = admin_mood_amount >= 0 ? TRUE : FALSE
+		SEND_SIGNAL(reader, COMSIG_ADD_MOOD_EVENT, "horoscope_admin", positive ? /datum/mood_event/horo_good : /datum/mood_event/horo_bad)
+		message_admins("[reader] has read the admin horoscope.")
```

Comment:
you should add to the game log here, since this is related to adminbussing and may be needed in the case of a staff report. same for the other horoscopes

Reply:
how do i add to game log?

Reply:
I think it's `log_game`